lib/auto_api/commands/windows_command.ex

```diff
@@ -0,0 +1,87 @@
+# AutoAPI
+# Copyright (C) 2018 High-Mobility GmbH
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see http://www.gnu.org/licenses/.
+#
+# Please inquire about commercial licensing options at
+# licensing@high-mobility.com
+defmodule AutoApi.WindowsCommand do
+  @moduledoc """
+  Handles Windows commands and apply binary commands on `%AutoApi.WindowsState{}`
+  """
+  @behaviour AutoApi.Command
+
+  alias AutoApi.WindowsState
+  alias AutoApi.WindowsCapability
+
+  @doc """
+  Parses the binary command and makes changes or returns the state
+
+  TODO: examples
+  """
+  @spec execute(WindowsState.t(), binary) :: {:state | :state_changed, WindowsState.t()}
+  def execute(%WindowsState{} = state, <<0x00>>) do
+    {:state, state}
+  end
+
+  def execute(%WindowsState{} = state, <<0x01, ws::binary>>) do
+    new_state = WindowsState.from_bin(ws)
+
+    if new_state == state do
+      {:state, state}
+    else
+      {:state_changed, new_state}
+    end
+  end
+
+  def execute(%WindowsState{} = _state, <<0x12, _payload::binary>>) do
+    # TODO
+    raise "Not implemented!"
```

Comment:
not sure if dialyzer likes this. `execute/2` callback is defined in `AutoApi.Command` and it always expects `{:ok, ..} | {:state_change, ..}`

Reply:
Hmm dialyzer gives a lot of errors all around the code in this branch so I didn't think about it at all :P

But, looking at it it only issues this complain:
```
lib/auto_api/states/windows_state.ex:32:pattern_match
The pattern
false

can never match the type
true
________________________________________________________________________________
```

Which it says for every State file, so I don't really know what the issue is. It points to the `use AutoApi.State` line